upstreams/container.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package upstreams
+
+import (
+	"sort"
+	"strings"
+
+	"github.com/blang/semver"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+
+	"sigs.k8s.io/zeitgeist/pkg/container"
+)
+
+// Container upstream representation
+type Container struct {
+	UpstreamBase `mapstructure:",squash"`
+	// Container URL, e.g. gcr.io/k8s-staging-kubernetes/conformance
+	URL string
+	// Optional: semver constraints, e.g. < 2.0.0
+	// Will have no effect if the dependency does not follow Semver
+	Constraints string
+}
+
+// LatestVersion returns the latest tag  for the given repository
+// (depending on the Constraints if set).
+func (upstream Container) LatestVersion() (string, error) { // nolint:gocritic
+	log.Debugf("Using Container flavour")
+	return latestContainerVersion(&upstream)
+}
+
+func latestContainerVersion(upstream *Container) (string, error) {
+	return latestContainerTag(upstream)
+}
+
+func latestContainerTag(upstream *Container) (string, error) {
```

Comment:
I'd suggest renaming this from "latestContainerTag" (etc..) for a few reasons:

1. `latest` is the default tag that gets pushed, so it's easy to confuse this with the literal `latest` tag.
2. "Latest" could mean "the most recently pushed tag" or "the highest semver-compatible tag that has ever been pushed". From scanning this code, I assume this should be the latter?

Something like "highestSemanticImageTag" is really explicit and would dispel some confusion.